lib/interface-types/src/decoders/binary.rs

```diff
@@ -234,10 +273,25 @@ fn types<'input, E: ParseError<&'input [u8]>>(
     let mut types = Vec::with_capacity(number_of_types as usize);
 
     for _ in 0..number_of_types {
-        consume!((input, inputs) = list(input, ty)?);
-        consume!((input, outputs) = list(input, ty)?);
+        consume!((input, type_kind) = byte(input)?);
```

Comment:
I don't understand how it could improve error messages. In all cases, it will happen in a loop. Maybe backtraces could be clearer, but still, this is the `types` parser, so it's helpful enough to avoid a refactoring for the moment I guess… ? 😅